recipes/aws-sdk-cpp/all/patches/0004-improve-pulseaudio-detection.patch

```diff
@@ -0,0 +1,56 @@
+diff --git a/aws-cpp-sdk-text-to-speech/CMakeLists.txt b/aws-cpp-sdk-text-to-speech/CMakeLists.txt
+index b1054515d5..82847accf5 100644
+--- a/aws-cpp-sdk-text-to-speech/CMakeLists.txt
++++ b/aws-cpp-sdk-text-to-speech/CMakeLists.txt
+@@ -9,8 +9,6 @@ file( GLOB TEXT_TO_SPEECH_HEADERS "include/aws/text-to-speech/*.h" )
+ 
+ set(PLATFORM_LIBS "")
+ 
+-check_include_files("pulse/simple.h" HAVE_PULSE)
+-
+ if(PLATFORM_WINDOWS)
+     file( GLOB TEXT_TO_SPEECH_PLATFORM_HEADERS "include/aws/text-to-speech/windows/*.h" )
+ 	file( GLOB TEXT_TO_SPEECH_PLATFORM_SOURCE "source/text-to-speech/windows/*.cpp" )
+@@ -20,13 +18,15 @@ if(PLATFORM_WINDOWS)
+ endif()
+ 
+ if (PLATFORM_LINUX)
+-    if(HAVE_PULSE)
++    find_package(pulseaudio)
++    if (${pulseaudio_FOUND})
+         file( GLOB TEXT_TO_SPEECH_PLATFORM_HEADERS "include/aws/text-to-speech/linux/*.h" )
+         file( GLOB TEXT_TO_SPEECH_PLATFORM_SOURCE "source/text-to-speech/linux/*.cpp" )
+ 
+         message(STATUS "Pulse audio header files have been detected, included pulse audio as a possible sound driver implementation.")
+         add_definitions("-DPULSE")
+-        set(PLATFORM_LIBS ${PLATFORM_LIBS} pulse pulse-simple)
++        include_directories(${pulseaudio_INCLUDES})
++        set(PLATFORM_LIBS ${PLATFORM_LIBS} ${pulseaudio_LIBRARIES})
```

Comment:
Prefer `pulseaudio::pulseaudio` imported target, it's more robust.

Usage of things like ${pulseaudio_INCLUDES} ${pulseaudio_LIBRARIES} are old CMake way to link a dependency.

Reply:
This would require a bit more refactoring, because the target is defined at the very end of the file. So you can't easily do `target_link_libraries` at this place.

Increasing the patch size will likely cause more maintenance burden going forward.